ws/session.nim

```diff
@@ -206,105 +206,70 @@ proc handleControl*(ws: WSSession, frame: Frame) {.async.} =
   else:
     raise newException(WSInvalidOpcodeError, "Invalid control opcode!")
 
-proc readFrame*(ws: WSSession): Future[Frame] {.async.} =
+proc readFrame*(ws: WSSession, maxSize: int = WSMaxMessageSize): Future[Frame] {.async.} =
   ## Gets a frame from the WebSocket.
   ## See https://tools.ietf.org/html/rfc6455#section-5.2
   ##
-
-  while ws.readyState != ReadyState.Closed:
-    let frame = await Frame.decode(
-      ws.stream.reader, ws.masked, ws.extensions)
-    debug "Decoded new frame", opcode = frame.opcode, len = frame.length, mask = frame.mask
-
-    # return the current frame if it's not one of the control frames
-    if frame.opcode notin {Opcode.Text, Opcode.Cont, Opcode.Binary}:
-      await ws.handleControl(frame) # process control frames# process control frames
-      continue
-
-    return frame
-
-proc ping*(ws: WSSession, data: seq[byte] = @[]): Future[void] =
-  ws.send(data, opcode = Opcode.Ping)
-
-proc recv*(
-  ws: WSSession,
-  data: pointer,
-  size: int): Future[int] {.async.} =
-  ## Attempts to read up to `size` bytes
-  ##
-  ## Will read as many frames as necessary
-  ## to fill the buffer until either
-  ## the message ends (frame.fin) or
-  ## the buffer is full. If no data is on
-  ## the pipe will await until at least
-  ## one byte is available
-  ##
-
-  var consumed = 0
-  var pbuffer = cast[ptr UncheckedArray[byte]](data)
+  var buf: array[0xFFFF, byte]
```

Comment:
We can't assume that we only work at message boundaries, some applications might benefit from receiving frames as they come.